src/frontend/src/utils/recoveryDeviceLabel.ts

```diff
@@ -0,0 +1,14 @@
+import { DeviceData } from "../../generated/internet_identity_types";
+
+export const recoveryDeviceToLabel = (
+  device: Omit<DeviceData, "alias">
```

Comment:
I think you can enforce that `device` has the correct `purpose` and `key_type` here and mostly avoid runtime errors

Reply:
I implemented that in commit `517e9e4f0c51d68828fa4029e087770018a83674` but it seems quite involved. Please have a look, maybe there is a way to simplify.